scripts/cmake/vcpkg_configure_make.cmake

```
@@ -232,7 +232,7 @@ macro(z_vcpkg_setup_detected_env env_name cmake_var)
         get_filename_component(cmd_PATH "${VCPKG_DETECTED_${cmake_var}}" DIRECTORY)
         get_filename_component(cmd_PROGRAM "${VCPKG_DETECTED_${cmake_var}}" NAME)
         set(ENV{${env_name}} "${cmd_PROGRAM}")

```

```suggestion
        vcpkg_host_path_list(REPEND "${cmd_PROGRAM}")
```

---

This will not be accepted since I don't know IF the full path is better. If that is confirmed, I would like to remove this part of code.

---

Typo in the suggestion: `PREPEND`

IMO in particular prepending to `PATH` is a problem because it may change which version of other tools is found and used, depending on the user's environment. (For mingw, it may trigger the mix of different runtimes which must be avoided.)

---

```suggestion
        set(ENV{${env_name}} "${VCPKG_DETECTED_${cmake_var}}")
```

FTR, that's what you propose @dg0yt ?
That should be more precise.

---

@mkuhn There was another line of comments where I wrote: "Why not set `ENV{${env_name}}` to the full path?" https://github.com/microsoft/vcpkg/pull/21847#discussion_r762811515.

---

Which would correspond to this if I'm not mistaken?

---

To be explicit, the suggestion by @JackBoosY is
- attached to the wrong line 
  - The `ENV{${env_name})` is to remain as is when used with `ENV{PATH}` changes.
  - `vcpkg_host_path_list` is to replace the explicit `ENV{PATH}` code.
- has a missing letter.

---

Ok, I will merge the suggestion in the next push.